Snail/Disposer.swift

```diff
@@ -7,16 +7,25 @@ public protocol DisposableType {
 }
 
 public class Disposer {
-    private(set) public var disposables: [DisposableType] = []
+    private(set) var disposables: [DisposableType] = []
+    private let disposablesQueue = DispatchQueue(label: "snail-disposer-queue")
```

Comment:
Should we use a lock here as well?

Reply:
yes, this makes sense. Disposables should only be mutated from a single thread at a time.